src/allreduce_robust.h

```diff
@@ -177,64 +199,91 @@ class AllreduceRobust : public AllreduceBase {
     // check point Ack, we use a two phase message in check point,
     // this is the second phase of check pointing
     static const int kCheckAck = 4;
+    // some node want to load cache
+    static const int kLoadCache = 8;
     // there are difference sequence number the nodes proposed
     // this means we want to do recover execution of the lower sequence
     // action instead of normal execution
-    static const int kDiffSeq = 8;
+    static const int kDiffSeq = 16;
```

Comment:
do we really need to add a bit? maybe make the new operation as 9, and just do XOR to check?

Reply:
those bits could exists at same time 2 | 4 | 8 | 16
XOR lose it's property as reducer might run multiple times on different nodes

Reply:
humm, a min

Reply:
I searched code base a bit, I didn't see the different bits in the flag can be set at the same time? can you point me to the code if there is any?

Reply:
okay, I am afraid we can't use 9, action summary will go through each bit and try to apply either & or | to get permutation of flag sets (1111, 1101 etc). 
      1
    10
  100
1000

About do XOR, I have same thoughts when I tried to find if all nodes are calling getcache at same time. Here is what it might happen, assume four nodes with load_cache bit <1, 1, 0, 0> the expected results would be 1. if we apply xor in reducer, it will produce 0 same as <0,0,0,0> or <1,1,1,1>